exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -252,6 +252,21 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 		return err
 	}
 
+	// Ensure that the kubernetes service of remote cluster has been created before apply ClusterResourceSet.
```

Comment:
Can you split this out into a separate function, reference the Kubernetes version it's required to support this behaviour for and add a TODO to remove it once that version is no longer supported by CAPI?